sklearn/ensemble/weight_boosting.py

```diff
@@ -145,7 +145,11 @@ def fit(self, X, y, sample_weight=None):
                 random_state)
 
             # Early termination
+            
             if sample_weight is None:
+                if estimator_error is not None and math.isnan(estimator_error):
+                    print("Early termination due to underflow of estimated_error.Iterations stopped")
```

Comment:
We don't use `print` statements for warning. (see [contributing guide](http://scikit-learn.org/stable/developers/contributing.html) for a warning example)

Or see this example in the code:
https://github.com/scikit-learn/scikit-learn/blob/3e85359f59bdb4ba2544afdc95a793cf0e60de7e/sklearn/preprocessing/data.py#L180-L185

More over if you introduce a if statement, the code is not tested therefore a test needs to be written.  See the test of the previous example here:
https://github.com/scikit-learn/scikit-learn/blob/3e85359f59bdb4ba2544afdc95a793cf0e60de7e/sklearn/preprocessing/tests/test_data.py#L219-L220

Reply:
Yes, I will change the print statement. And regarding writing the test,
1. Can I use some imported datasets as tests, from  sklearn.datasets?
2.How many tests should I include?

Thanks

Reply:
Thanks! I'll work on the test

Reply:
@massich A separate function from testing it would suit?
Heres a snapshot. Any suggestions would be great,
![ada_f](https://user-images.githubusercontent.com/20499290/32977646-937a094a-cc57-11e7-84c5-c9bdbf8d6ad4.jpg)

Reply:
If there's no place where something similar is tested, create a new function. But I would use a more descriptive name like `test_early_stop_when_estimator_error_bcomes_nan`. I would also add a reference to the original issue. 

To avoid PEP8 errors, I would also set the warning message into a variable to further compare to. (And make sure that the warning raised and catch are the same (try to find a name bettern than `w` or `early_stop_warnmsg`)

```
early_stop_warnmsg = "Early termination due to underflow of estimated_error. Iterations stopped"
assert_warns_message(UserWarning, early_stop_warnsg, clf.fit, iris.data, iris.target)
```

Reply:
Great. Thanks, I will make these changes.

Reply:
I don't think the estimator error is underflowing. Then won't estimated error be approximated to zero? I think the sample weights are shooting up to infinite when summed. Causing inf/inf division for estimated error. Giving a nan. Any suggestions?